drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -251,12 +253,34 @@ static int adc_smartbond_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
+#ifdef CONFIG_PM_DEVICE
+static int gpadc_smartbond_pm_action(const struct device *dev,
+				   enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+	    da1469x_pd_acquire(MCU_PD_DOMAIN_PER);
+		GPADC->GP_ADC_CTRL_REG = GPADC_GP_ADC_CTRL_REG_GP_ADC_EN_Msk;
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		GPADC->GP_ADC_CTRL_REG &= ~(GPADC_GP_ADC_CTRL_REG_GP_ADC_EN_Msk);
+		da1469x_pd_release(MCU_PD_DOMAIN_PER);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
 static int adc_smartbond_init(const struct device *dev)
 {
 	int err;
 	struct adc_smartbond_data *data = dev->data;
 	const struct adc_smartbond_cfg *config = dev->config;
 
+	da1469x_pd_acquire(MCU_PD_DOMAIN_PER);
```

Comment:
There should be provision for device runtime (for reference you can take a loot at [SPI](https://github.com/zephyrproject-rtos/zephyr/pull/73008/files#diff-8a70974e4b68b47cb59e0ba6253847031c56e6b55dc21e273a9cebc72e887b66R364)). As such, GPADC registers and optionally pins settings should be encapsulated in a subroutine namely `gpadc_smartbond_resume`.

Reply:
@NiekIlmer, kind reminder to Add support for the device runtime scheme.

Reply:
Missed the changes here, fixed now